drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/rightpanel/ListGroupItemPresenter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.rightpanel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import com.google.gwt.dom.client.DivElement;
+import org.drools.workbench.screens.scenariosimulation.client.models.FactModelTree;
+
+@Dependent
+public class ListGroupItemPresenter implements ListGroupItemView.Presenter {
+
+    @Inject
+    private Instance<ListGroupItemView> instance;
+
+    @Inject
+    FieldItemPresenter fieldItemPresenter;
+
+    @Inject
+    RightPanelPresenter rightPanelPresenter;
+
+    List<ListGroupItemView> listGroupItemViewList = new ArrayList<>();
+
+    @Override
+    public DivElement getDivElement(String factName, FactModelTree factModelTree) {
+        ListGroupItemView listGroupItemView = commonGetListGroupItemView(false);
+        populateListGroupItemView(listGroupItemView, factName, factModelTree);
+        return listGroupItemView.getDivElement();
+    }
+
+    @Override
+    public DivElement getDivElement(String factName, String factModelTreeClass) {
+        ListGroupItemView listGroupItemView = commonGetListGroupItemView(true);
+        populateListGroupItemView(listGroupItemView, factName, factModelTreeClass);
+        return listGroupItemView.getDivElement();
+    }
+
+    @Override
+    public void toggleRowExpansion(ListGroupItemView listGroupItemView, boolean currentlyShown) {
+        if (listGroupItemViewList.contains(listGroupItemView)) {
+            if (currentlyShown) {
+                listGroupItemView.closeRow();
+            } else {
+                if (listGroupItemView.isToExpand()) {
+                    FactModelTree factModelTree = rightPanelPresenter.getFactModelTree(listGroupItemView.getFactType());
+                    populateListGroupItemView(listGroupItemView, listGroupItemView.getFactName(), factModelTree);
+                    listGroupItemView.setToExpand(false);
+                }
+                listGroupItemView.expandRow();
+            }
+        }
+    }
+
+    protected ListGroupItemView getListGroupItemView() {  // This is needed for test because Mockito can not mock Instance
+        return instance.get();
```

Comment:
Again if this just provides the Presenters rename to PresenterProvider

Reply:
I have implemented a ViewsProvider to separate "instantiation" logic from presenting one

[DROOLS-2909] Implemented ViewsProvider to clearly separate "provider" logic from presenter one